internal/nginx-meshctl/commands/upgrade.go

```diff
@@ -233,8 +252,9 @@ func (u *upgrader) upgrade(version string) error {
 // - copy on top of the new release's deploy-time configuration
 // - get previous release's run-time configuration (mesh-config ConfigMap)
 // - copy on top of the new release's deploy-time configuration
-// - set new version.
-func (u *upgrader) buildValues(ctx context.Context, version string) (map[string]interface{}, error) {
+// - set new version
+// - set new registry server if needed.
+func (u *upgrader) buildValues(ctx context.Context, version string, registry string) (map[string]interface{}, error) {
```

Comment:
nit: since version and registry are the same type, you only have to specify `string` once.